lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -225,6 +227,8 @@ def add_payment_urls(post, options)
       end
 
       def add_customer_object(post, payment, options)
+        return unless payment.instance_of?(CreditCard)
+
         post[:name] = "#{payment.first_name} #{payment.last_name}"
         post[:phone_number] = options[:billing_address][:phone].gsub(/\D/, '') if options[:billing_address]
         post[:email] = options[:email] if options[:email]
```

Comment:
I saw that `email` isn't required, just recommended, but if an `email` field isn't explicitly included in the auth/purchase request, doesn't it need to be pulled from the payment method like the `name`, where as `[:billing_address]` is populated by Core?

Reply:
It seems like `options[:email]` is pretty common throughout AM, but there are instances of email being pulled from the `address` hash. I don't see any instance of `email` coming from the payment method, however.